Source/cg_PresetsManager.cpp

```diff
@@ -92,97 +92,94 @@ bool PresetsManager::forceLoad(int const presetNumber)
-
-        auto const * presetData{ *maybe_presetData };
-
-        if (presetData->hasAttribute ("numberOfSources"))
-            mSources.setSize(presetData->getIntAttribute("numberOfSources"));
-
-        if (presetData->hasAttribute("firstSourceId"))
-            mFirstSourceId = SourceId{ presetData->getIntAttribute("firstSourceId") };
-
-        // Store the preset's source positions in a new SourcesSnapshots
-        SourcesSnapshots snapshots{};
-        for (auto & source : mSources) {
-            SourceSnapshot snapshot{};
-            auto const index{ source.getIndex() };
-            auto const xPosId{ getFixedPosSourceName(FixedPositionType::initial, index, 0) };
-            auto const yPosId{ getFixedPosSourceName(FixedPositionType::initial, index, 1) };
-            if (presetData->hasAttribute(xPosId) && presetData->hasAttribute(yPosId)) {
-                juce::Point<float> const normalizedInversedPosition{
-                    static_cast<float>(presetData->getDoubleAttribute(xPosId)),
-                    static_cast<float>(presetData->getDoubleAttribute(yPosId))
-                };
-                auto const inversedPosition{ normalizedInversedPosition * 2.0f - juce::Point<float>{ 1.0f, 1.0f } };
-                juce::Point<float> const position{ inversedPosition.getX(), inversedPosition.getY() * -1.0f };
-                snapshot.position = position;
-                auto const zPosId{ getFixedPosSourceName(FixedPositionType::initial, index, 2) };
-                if (presetData->hasAttribute(zPosId)) {
-                    auto const elevation {getFixedPosSourceName(FixedPositionType::initial, index, 2)};
-                    auto const inversedNormalizedElevation{ static_cast<float>(presetData->getDoubleAttribute(elevation)) };
-                    snapshot.z = MAX_ELEVATION * (1.0f - inversedNormalizedElevation);
-                }
-            }
-            if (source.isPrimarySource()) {
-                snapshots.primary = snapshot;
-            } else {
-                snapshots.secondaries.add(snapshot);
-            }
-        }
+    if (presetNumber <= 0)
+        return false;
 
-        //load the snapshots into the enforcers, which are references to the ControlGrisAudioProcessor members
-        mPositionLinkEnforcer.loadSnapshots(snapshots);
-        if (mSources.getPrimarySource().getSpatMode() == SpatMode::cube) {
-            mElevationLinkEnforcer.loadSnapshots(snapshots);
-        }
+    auto const maybe_presetData{ getPresetData(presetNumber) };
+    if (!maybe_presetData.has_value()) {
+        return false;
+    }
 
-        auto const xTerminalPositionId{ getFixedPosSourceName(FixedPositionType::terminal, SourceIndex{ 0 }, 0) };
-        auto const yTerminalPositionId{ getFixedPosSourceName(FixedPositionType::terminal, SourceIndex{ 0 }, 1) };
-        auto const zTerminalPositionId{ getFixedPosSourceName(FixedPositionType::terminal, SourceIndex{ 0 }, 2) };
+    load(**maybe_presetData);
```

Comment:
I'm not too familiar with this api but since there's a double dereference, shouldn't the result of ` *maybe_presetData` be checked for NULL too ?